File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/shape/ShapeType.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.shape;

import android.graphics.drawable.GradientDrawable;

import java.util.Locale;

import androidx.annotation.NonNull;
import androidx.annotation.RestrictTo;

import com.urbanairship.json.JsonException;

/**
 * Shape types.
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public enum ShapeType {
    RECTANGLE("rectangle", GradientDrawable.RECTANGLE),
    ELLIPSE("ellipse", GradientDrawable.OVAL);

    @NonNull
    private final String value;
    private final int drawableShape;

    ShapeType(@NonNull String value, int drawableShape) {
        this.value = value;
        this.drawableShape = drawableShape;
    }

    @NonNull
    public static ShapeType from(@NonNull String value) throws JsonException {
        for (ShapeType type : ShapeType.values()) {
            if (type.value.equals(value.toLowerCase(Locale.ROOT))) {
                return type;
            }
        }
        throw new JsonException("Unknown ShapeType value: " + value);
    }

    public int getDrawableShapeType() {
        return drawableShape;
    }

    @NonNull
    @Override
    public String toString() {
        return name().toLowerCase(Locale.ROOT);
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/shape/ShapeType.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.shape

import android.graphics.drawable.GradientDrawable
import androidx.annotation.RestrictTo
import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue

/**
 * Shape types.
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public enum class ShapeType(
    private val value: String,
    public val drawableShape: Int
) {

    RECTANGLE("rectangle", GradientDrawable.RECTANGLE),
    ELLIPSE("ellipse", GradientDrawable.OVAL);

    override fun toString(): String {
        return name.lowercase()
    }

    public companion object {

        @Throws(JsonException::class)
        public fun from(value: JsonValue): ShapeType {
            val content = value.requireString().lowercase()

            return entries.firstOrNull { it.value == content }
                ?: throw JsonException("Unknown ShapeType value: $value")
        }
    }
}
